File: 14-Longest-Common-Prefix/solution.c
char* longestCommonPrefix(char** strs, int strsSize) {
    char * str = NULL;
    if (strsSize < 1)
    {
        return "";
    }
    str = strs[0];
    size_t len = strlen(str);
    for (int i = 1; i < strsSize; i++)
    {
        for (size_t j = 0; j < len; j++)
        {
            if (strs[i][j] == '\0' || str[j] != strs[i][j])
            {
                len = j;
                break;
            }
        }
    }

    if (len > 0)
    {
        str[len] = '\0';
        return str;
    }
    return "";
}

File: 14-Longest-Common-Prefix/solution.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ""

        for i in xrange(len(strs[0])):
            for string in strs:
                if i >= len(string) or string[i] != strs[0][i]:
                    return strs[0][:i]
        return strs[0]